src/utilities/request.jl

```diff
@@ -206,16 +236,28 @@ function _http_request(http_backend::HTTPBackend, request::Request, response_str
                 response_stream=buffer,
                 http_options...,
             )
-        catch e
+
+            # We'll rely on lexical scoping; `buffer` and `response`
+            # are bindings in the outer scope, so we don't need to return here.
+            return nothing
+        end
+
+    check = function (s, e)
             # `Base.IOError` is needed because HTTP.jl can often have errors that aren't
             # caught and wrapped in an `HTTP.IOError`
             # https://github.com/JuliaWeb/HTTP.jl/issues/382
-            @delay_retry if isa(e, Sockets.DNSError) ||
-                isa(e, HTTP.ParseError) ||
-                isa(e, HTTP.IOError) ||
-                isa(e, Base.IOError) ||
-                (isa(e, HTTP.StatusError) && _http_status(e) >= 500)
-            end
+            return isa(e, Sockets.DNSError) ||
+                   isa(e, HTTP.ParseError) ||
+                   isa(e, HTTP.IOError) ||
+                   isa(e, Base.IOError) ||
+                   (isa(e, HTTP.StatusError) && _http_status(e) >= 500)
+        end
+
+    delays=AWSExponentialBackoff(; max_attempts=4)
+    
+    try
+        retry(; check, delays) do
+            get_response()   # Inline this function
         end
```

Comment:
@omus I think we would need
```suggestion
        retry(; check, delays) do
            get_response()   # Inline this function
        end()
```
or
```suggestion
        retry(get_response; check, delays)()
```
still, since the `do`-block is just constructing `retry(() -> get_response(); check, delays)` here. (Also, we need to use the old keyword arg syntax or `@compat`)